mods/auto-hide-desktop-after-inactivity.wh.cpp

```diff
@@ -0,0 +1,327 @@
+
+bool IsDesktopVisible() {
+    HWND shellWindow = GetShellWindow();
+    HWND foreground = GetForegroundWindow();
+    return foreground == shellWindow;
+}
+
+void ToggleDesktopWithWinD() {
+    keybd_event(VK_LWIN, 0, 0, 0);           // Win down
+    keybd_event('D', 0, 0, 0);               // D down
+    keybd_event('D', 0, KEYEVENTF_KEYUP, 0); // D up
+    keybd_event(VK_LWIN, 0, KEYEVENTF_KEYUP, 0); // Win up
+}
+
+void HideDesktop() {
+    if (!IsDesktopVisible()) {
+        ToggleDesktopWithWinD();
+    }
+    GetCursorPos(&lastMousePos);
+    desktopShownByMod = true;
+    timeDesktopShown = GetTickCount();
+    lockPending = (setting_lockTimeoutSeconds > 0);
+}
+
+void RestoreWindows() {
+    if (IsDesktopVisible()) {
+        ToggleDesktopWithWinD();
+    }
+
+    desktopShownByMod = false;
+}
+
+void LoadSettings() {
+    int value;
+    value = Wh_GetIntSetting(L"timeoutSeconds");
+    setting_timeoutSeconds = (value > 0) ? value : 300;
+
+    value = Wh_GetIntSetting(L"lockTimeoutSeconds");
+    setting_lockTimeoutSeconds = (value >= 0) ? value : 0;
+
+    setting_ignoreFullscreenApps = Wh_GetIntSetting(L"ignoreFullscreenApps");
+
+    const wchar_t* excludeList = Wh_GetStringSetting(L"excludeProcesses");
+    excludedProcessList.clear();
+    if (excludeList && *excludeList) {
+        std::wstring listStr(excludeList);
+        size_t pos = 0;
+        while ((pos = listStr.find(L';')) != std::wstring::npos) {
+            std::wstring token = listStr.substr(0, pos);
+            if (!token.empty())
+                excludedProcessList.push_back(token);
+            listStr.erase(0, pos + 1);
+        }
+        if (!listStr.empty())
+            excludedProcessList.push_back(listStr);
+    }
+
+    Wh_Log(L"Settings loaded: timeoutSeconds = %d, lockTimeoutSeconds = %d",
+        setting_timeoutSeconds, setting_lockTimeoutSeconds);
+}
+
+bool IsExcludedProcessRunning() {
+    HWND hwnd = GetForegroundWindow();
+    if (!hwnd)
+        return false;
+
+    DWORD pid;
+    GetWindowThreadProcessId(hwnd, &pid);
+
+    HANDLE hProcess = OpenProcess(PROCESS_QUERY_LIMITED_INFORMATION, FALSE, pid);
+    if (!hProcess)
+        return false;
+
+    wchar_t exePath[MAX_PATH];
+    DWORD size = ARRAYSIZE(exePath);
+    bool result = false;
+
+    if (QueryFullProcessImageNameW(hProcess, 0, exePath, &size)) {
+        std::wstring exeName = exePath;
+        size_t pos = exeName.find_last_of(L"\\/");
+        if (pos != std::wstring::npos)
+            exeName = exeName.substr(pos + 1);
+
+        Wh_Log(L"Foreground exe: %s", exeName.c_str()); // ADD THIS
+
+        for (const auto& excluded : excludedProcessList) {
+            if (_wcsicmp(exeName.c_str(), excluded.c_str()) == 0) {
+                result = true;
+                break;
+            }
+        }
+    }
+
+    CloseHandle(hProcess);
+    return result;
+}
+
+bool IsForegroundWindowFullscreen() {
+    HWND hwnd = GetForegroundWindow();
```

Comment:
There's a WinAPI function which allows to query the fullscreen state:
https://github.com/ramensoftware/windhawk-mods/blob/9d7df8f53b5d834fbed6034c7c7a2dd03033f66a/mods/edge-hot-corner-desktop-switch.wh.cpp#L69-L83